jerry-core/ecma/builtin-objects/ecma-builtin-array-prototype.c

```diff
@@ -1958,9 +1972,19 @@ ecma_builtin_array_prototype_object_every (ecma_value_t this_arg, /**< this argu
         ECMA_TRY_CATCH (call_value, ecma_op_function_call (func_object_p, arg2, call_args, 3), ret_value);
 
         /* 7.c.iii */
-        if (!ecma_op_to_boolean (call_value))
+        if (mode == ARRAY_ROUTINE_EVERY)
         {
-          ret_value = ECMA_VALUE_FALSE;
+          if (!ecma_op_to_boolean (call_value))
+          {
+            ret_value = ECMA_VALUE_FALSE;
+          }
+        }
+        else if (mode == ARRAY_ROUTINE_SOME)
+        {
+          if (ecma_op_to_boolean (call_value))
+          {
+            ret_value = ECMA_VALUE_TRUE;
+          }
```

Comment:
This `if` structure can be merged like this: 
```c
if (mode == ARRAY_ROUTINE_EVERY || mode == ARRAY_ROUTINE_SOME)
{
    ret_value = ecma_op_to_boolean (call_value) ? ECMA_VALUE_TRUE : ECMA_VALUE_FALSE;
}
```
Therefore `ecma_op_to_boolean (call_value)` is performed only once in both cases.

Reply:
Ah there's a small issue actually, there are cases when the `ret_value` shouldn't change. The `if` structure can still be reduced to
```c
        if (mode == ARRAY_ROUTINE_EVERY && !ecma_op_to_boolean (call_value))
        {
            ret_value = ECMA_VALUE_FALSE;
        }
        else if (mode == ARRAY_ROUTINE_SOME && ecma_op_to_boolean (call_value))
        {
            ret_value = ECMA_VALUE_TRUE;
        }